runtime/compiler/optimizer/J9RecognizedCallTransformer.cpp

```diff
@@ -306,6 +306,16 @@ void J9::RecognizedCallTransformer::process_java_lang_StringLatin1_inflate_BIBII
 
    TR_ASSERT_FATAL(comp()->cg()->getSupportsArrayTranslateTROTNoBreak(), "Support for arraytranslateTROTNoBreak is required");
 
+   // Anchor a copy of the call node just before treetop so that all of the
+   // children will be commoned across the split point, and all of the temps
+   // will be initialized before the first opportunity to go to the fallback
+   // path. Otherwise, the fallback path could end up using temps that are
+   // sometimes uninitialized. This copy will be removed just after splitting.
+   TR::TreeTop *callCopyTT = TR::TreeTop::create(
+      comp(), TR::Node::create(node, TR::treetop, 1, node->duplicateTree(false)));
```

Comment:
I have to say that I like this way of handling the problem.  Some other places in `J9RecognizedCallTransformer` and `Value Propagation` use calls to `TransformUtil::createTempsForCall` prior to splitting blocks, but doing that in combination with the `fixUpCommoning` of the block splitter might result in more temporaries than is really necessary.